docs/udf.rst

```diff
@@ -0,0 +1,367 @@
+Each annotated method is a factory for an invocable aggregate function in KSQL. The annotation only has a single
+field ``description`` that is required. You can use this to better describe what a particular version
+of the UDF does, for example:
+
+.. code:: java
+
+    @UdafFactory(description = "Sums BIGINT columns.")
+    public static TableUdaf<Long, Long> createSumLong(){...}
+
+    @UdafFactory(description = "Sums the length of VARCHAR columns".)
+    public static Udaf<String, Long> createSumLengthString(final String initialString)
+
+
+===============
+Supported Types
+===============
+
+The types supported by UDFs are currently limited to:
+
++--------------+------------------+
+|  Java Type   | KSQL Type        |
++==============+==================+
+| int          | INTEGER          |
++--------------+------------------+
+| Integer      | INTEGER          |
++--------------+------------------+
+| boolean      | BOOLEAN          |
++--------------+------------------+
+| Boolean      | BOOLEAN          |
++--------------+------------------+
+| long         | BIGINT           |
++--------------+------------------+
+| Long         | BIGINT           |
++--------------+------------------+
+| double       | DOUBLE           |
++--------------+------------------+
+| Double       | DOUBLE           |
++--------------+------------------+
+| String       | VARCHAR          |
++--------------+------------------+
+| List         | ARRAY            |
++--------------+------------------+
+| Map          | MAP              |
++--------------+------------------+
+
+Note: Complex types other than List and Map are not currently supported
+
+
+=========
+Deploying
+=========
+
+To deploy your UDFs you need to create a jar containing all of the classes required by the UDFs.
+If you depend on third-party libraries then this should be an uber-jar containing those libraries.
+Once the jar is created you need to deploy it to each KSQL server instance. The jar should be copied
+to the ``ext/`` directory that is part of the KSQL distribution. The ``ext/`` directory can be configured
+via the ``ksql.extension.dir``.
+
+The jars in the ``ext`` directory are only scanned at start-up, so you will need to restart your
+KSQL server instances to pick up new UDFs.
+
+
+=====
+Usage
+=====
+
+Once your UDFs are deployed you can call them in the same way you would invoke any of the KSQL
+built-in functions. The function names are case-insensitive. For example, using the ``multiply`` example above:
+
+.. code:: sql
+
+    CREATE STREAM number_stream (int1 INT, int2 INT, long1 BIGINT, long2 BIGINT)
+      WITH (VALUE_FORMAT = 'JSON', KAFKA_TOPIC = 'numbers');
+
+    SELECT multiply(int1, int2), MULTIPLY(long1, long2) FROM number_stream;
+
+
+
+============
+Blacklisting
+============
+
+You can blacklist classes and packages such that they can't be used from a UDF. There is small
```

Comment:
Please add 1-2 lines to describe WHY the user would even care.

Like: "Depending on the deployment environment and security requirements, the use of UDFs may pose security risks because UDFs represent third-party code that are executed by KSQL servers. To reduce the attack surface of KSQL UDFs, you can optionally blacklist Java classes and packages such that they can't be used from a UDF.  (You can also disable UDF functionality completely, see next section [blabla-link].)"